File: java/christmas/src/test/java/christmas/DiscountTest.java
package christmas;
import christmas.Model.Discount;
import christmas.Model.Order;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class DiscountTest {

    @Test
    void calculateDdayDiscount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 20);

        int ddayDiscount = discount.calculateDdayDiscount();

        assertEquals(2900, ddayDiscount);
    }

    @Test
    void calculateSpecialDiscount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 25);

        int specialDiscount = discount.calculateSpecialDiscount();

        assertEquals(1000, specialDiscount);
    }

    @Test
    void calculateFreebieDiscount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 20);

        int freebieDiscount = discount.calculateFreebieDiscount();

        assertEquals(0, freebieDiscount);
    }

    @Test
    void calculateWeekDayDiscount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 19);

        int weekDayDiscount = discount.calculateWeekDayDiscount();

        assertEquals(0, weekDayDiscount);
    }

    @Test
    void calculateWeekendDiscount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 20);

        int weekendDiscount = discount.calculateWeekendDiscount();

        assertEquals(0, weekendDiscount);
    }

    @Test
    void menuCount() {
        Order mockOrder = new Order("티본스테이크-2,제로콜라-1,양송이수프-1");
        Discount discount = new Discount(mockOrder, 20);

        int dessertCount = discount.menuCount("디저트");

        assertEquals(0, dessertCount);
    }
}

File: kotlin/christmas/src/test/kotlin/christmas/DiscountTest.kt
package christmas
import christmas.model.discount
import christmas.model.order
import christmas.view.ErrorView
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.*

class DiscountTest {

    @Test
    fun `calculateDdayDiscount should return correct discount for D-day`() {
        val order = order("양송이수프-2,타파스-1,티본스테이크-3")
        val discount = discount(order, 25) // D-day is 25th
        assertEquals(3400, discount.calculateDdayDiscount())
    }

    @Test
    fun `calculateDdayDiscount should return 0 discount after D-day`() {
        val order = order("양송이수프-2,타파스-1,티본스테이크-3")
        val discount = discount(order, 26) // After D-day
        assertEquals(0, discount.calculateDdayDiscount())
    }

    @Test
    fun `calculateSpecialDiscount should return 1000 discount on special day`() {
        val order = order("양송이수프-2,타파스-1,티본스테이크-3")
        val discount = discount(order, 25) // Special day
        assertEquals(1000, discount.calculateSpecialDiscount())
    }

    @Test
    fun `calculateSpecialDiscount should return 0 discount on non-special day`() {
        val order = order("양송이수프-2,타파스-1,티본스테이크-3")
        val discount = discount(order, 26) // Non-special day
        assertEquals(0, discount.calculateSpecialDiscount())
    }

    @Test
    fun `calculateFreebieDiscount should return 25000 discount when freebie is true`() {
        val order = order("양송이수프-2,타파스-1,티본스테이크-3")
        val discount = discount(order, 25)
        assertTrue(discount.calculateFreebieDiscount() == 25000)
    }

    @Test
    fun `calculateFreebieDiscount should return 0 discount when freebie is false`() {
        val order = order("양송이수프-2,타파스-1,제로콜라-3")
        val discount = discount(order, 25)
        assertTrue(discount.calculateFreebieDiscount() == 0)
    }

    // Add more tests for other methods in the Discount class as needed.
}
